packages/config/src/project/index.ts

```diff
@@ -105,9 +105,17 @@ export function updateWithPluginValues (cfg: FullConfig, modifiedConfig: any, te
     debug('resolved config object %o', cfg.resolved)
   }
 
+  const diffsClone = _.cloneDeep(diffs)
+
   // merge cfg into overrides
   const merged = _.defaultsDeep(diffs, cfg)
 
+  for (const key in diffsClone) {
+    if (Array.isArray(diffsClone[key])) {
+      merged[key] = _.cloneDeep(diffsClone[key])
+    }
+  }
+
```

Comment:
Hello @lmiller1990. I made the suggested changes and locally tried the original reproduction repo, and it works perfectly fine for it. However, I was not able to run the tests locally. After selecting the spec, I was met with a lot of errors. I modified the `it()` block under which you had added these tests to `it.only()`, however it gave me the following error.

```
cy.task('__internal_withCtx') timed out after waiting 60000ms.



Because this error occurred during a before each hook we are skipping the remaining tests in the current suite: specChange subscription
```